dask_match/expr.py

```diff
@@ -128,6 +128,9 @@ def dependencies(self):
         # Dependencies are `Expr` operands only
         return [operand for operand in self.operands if isinstance(operand, Expr)]
 
+    def _layer(self):
+        return {(self._name, i): self._task(i) for i in range(self.npartitions)}
```

Comment:
I may be on board with this if `Expr._task` is defined with a clear `NotImplementedError` message.